### PR TITLE
Collapsables

### PR DESCRIPTION
**Pull request by troyschultz-ucsc**
allow users to add a photo of a recipe they have cooked onto their favorites cards   by a direct image url entry from the user.

### CODE REVIEW

File: .DS_Store
Comment (CarsonWebster) [inline .DS_Store:1]: No need to push .DS_Store. I thought this was in the .gitignore?

File: controllers.py [hunk 1]
```diff
@@ -287,23 +281,41 @@ def getFavs():
 def togglePin():
     userID = auth.current_user.get("id")
     favID = request.json.get("favID")
+    imageUrl = request.json.get("imageUrl")
```
Comment (CarsonWebster) [inline controllers.py:284]: Should we really need to pass an imageUrl through the url when we call togglePin? Idk if this is the best solution to updating the imageUrl? Where does this url come from? Modifying the recipes image shouldn't be paired with pinning functionality

File: controllers.py [hunk 2]
```diff
@@ -318,6 +330,11 @@ def getPinned():
             "pinned": recipe.pinned,
             "user_name": first_name,
         }
+        if recipe.pinned:
+            recipe_dict["imageUrl"] = recipe.imageUrl
+            recipe.update_record(imageUrl=recipe.imageUrl)
```
Comment (CarsonWebster) [inline controllers.py]: What's going on here? Of the pinned recipe rows, are we getting the existing imageUrl, adding it to recipe in local recipe_dict variable, and updating the recipe in the db with that same Url? 

This seems redundant and we should't be updating db rows every time it's queried

**Review by CarsonWebster — DISMISSED** [review]
I'm not sure this is quite functional yet. Let's see if we can rework the parts where I left comments. 

Until we get image or URL uploading to the database from the front end, let's refrain from merging to main

**Comment by troyschultz-ucsc** [comment]
> I'm not sure this is quite functional yet. Let's see if we can rework the parts where I left comments.
> 
> Until we get image or URL uploading to the database from the front end, let's refrain from merging to main

Thank you for reviewing my code. I appreciate your feedback. I understand that there are some comments that need to be addressed. If you have a moment, could you kindly assist me with the specific points you mentioned? I worked on this code all night, and I'm feeling a bit drained at the moment. While it may not be perfect, this version is at least functional and allows users to hyperlink their own images instead of storing the image files on the server. I would be grateful for your assistance in resolving the remaining issues. Thank you for your understanding and support.

**Comment by CarsonWebster** [comment]
Refactored the image URL structure :)

**Review by CarsonWebster — APPROVED** [review]
Approve now can I fix merge conflicts?

**Pull request merged**